### ISSUE TITLE
Nonexistant source file crashes the compiler

### ISSUE THREAD

**Issue by tschaer** [tschaer/Oberon-0-go#2]
If the compiler is invoked with a filename it can't find, it will panic with a null pointer dereference.

The problem is here in `OSS.Init()`:
 
     /* Scanner init */
     func Init(fname string) {
	     file, err := os.Open(fname)
	     if err != nil {
		     fmt.Println(err)
	     } else {
		     r = bufio.NewReader(file)
		     ch, _ = r.ReadByte()
		     Error = false
	     }
     }

The error is correctly identifed, but not handled. The compiler should exit after failing to open the file.

**Comment by tschaer** [comment]
Need to move this up the call chain

**Comment by tschaer** [comment]
Issue is fixed in commit 6208f75 by moving the file opening code into the top level module:

```go
   filename := os.Args[1]
   file, err := os.Open(filename)
   if err != nil {
      fmt.Println(err)
      return
   } else {
      OSP.Compile(bufio.NewReader(file))
   }
```

Now if os.Open() returns err, the program is aborted.

Requires changes to OSP.Compile() and OSS.Init() to accept a `bufio.Reader` object instead of the filename contained in a string.
